src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -726,18 +797,127 @@ bool InputStateMachineEngine::_IsModified(const unsigned short cParams)
 // - modifierParam - the VT modifier value to convert
 // Return Value:
 // - The equivalent INPUT_RECORD modifier value.
-DWORD InputStateMachineEngine::_GetModifier(const unsigned short modifierParam)
+DWORD InputStateMachineEngine::_GetModifier(const unsigned short modifierParam, bool SGRMode)
 {
-    // VT Modifiers are 1+(modifier flags)
-    unsigned short vtParam = modifierParam - 1;
-    DWORD modifierState = modifierParam > 0 ? ENHANCED_KEY : 0;
+    unsigned short vtParam = modifierParam;
+    DWORD modifierState = 0;
+    if (!SGRMode)
+    {
+        // VT Modifiers are 1+(modifier flags)
+        vtParam = modifierParam - 1;
+        modifierState = modifierParam > 0 ? ENHANCED_KEY : 0;
+    }
 
     bool fShift = WI_IsFlagSet(vtParam, VT_SHIFT);
     bool fAlt = WI_IsFlagSet(vtParam, VT_ALT);
     bool fCtrl = WI_IsFlagSet(vtParam, VT_CTRL);
     return modifierState | (fShift ? SHIFT_PRESSED : 0) | (fAlt ? LEFT_ALT_PRESSED : 0) | (fCtrl ? LEFT_CTRL_PRESSED : 0);
 }
 
+// Method Description:
+// - Synthesize the button state for the Mouse Input Record from an SGR VT Sequence
+// - Here, we refer to and maintain the global state of our mouse.
+// - Mouse wheel events are added at the end no keep them out of the global state
+// Arguments:
+// - wch: the wchar_t representing whether the button was pressed or released
+// - rgusParams: the wchar_t to get the mapped vkey of. Represents the direction of the button (down vs up)
+// - pdwButtonState: Recieves the button state for the record
+// Return Value:
+// true iff we were able to synthesize pdwButtonState
+bool InputStateMachineEngine::_UpdateSGRMouseButtonState(const wchar_t wch,
+                                                         _In_reads_(cParams) const unsigned short* const rgusParams,
+                                                         _Out_ DWORD* const pdwButtonState,
+                                                         _Out_ DWORD* const pdwEventFlags)
+{
+    *pdwButtonState = _mouseButtonState;

```

Comment:
Probably add a comment suggesting that we're starting with the state from the last mouse event we received